broker/src/scheduler/active_calls_gc.erl

```diff
@@ -24,9 +29,36 @@ handle_call(_Request, _From, State) ->
 handle_cast(_Msg, State) ->
   {noreply, State}.
 
+notify_failed_to_callback_url(Call) ->
+  case Call#call_log.callback_url of
+    undefined -> ok;
+    <<>> -> ok;
+    Url ->
+      CallSid = util:to_string(Call#call_log.id),
+      SessionVars = case Call#call_log.js_context of
+        "" -> [];
+        undefined -> [];
+        _ -> session:session_vars(Call#call_log.js_context)
+      end,
+      spawn(fun() ->
+        Uri = uri:parse(binary_to_list(Url)),
+        QueryString = [{"CallSid", CallSid}, {"CallStatus", "failed"}, {"CallDuration", 0} | SessionVars],
+        MergedQueryString = Uri#uri.query_string ++ QueryString,
+        NewUri = Uri#uri{query_string = MergedQueryString},
+        NewUri:get([{full_result, false}])
+      end)
+  end.
+
 handle_info(cancel_active_calls, State) ->
   N = verboice_config:minutes_for_cancelling_active_calls(),
   N_Minutes_Ago = util:seconds_ago(N * 60),
+  
+  % First the callback url is notified
+  lists:foreach(fun(Call) ->
+      notify_failed_to_callback_url(Call)
+    end, call_log:calls_started_before(N_Minutes_Ago)),
+  
+  % Then the call logs are updated to mark them as failed
   Count = call_log:cancel_active_calls_started_before(N_Minutes_Ago),
```

Comment:
This feature is all about consistency - we're trying to ensure that, no matter what happens to a call, the app that's handling the call gets to know its state.

So I wouldn't split the "notify callback" and "mark call as canceled" parts like we're doing.

Since we're already going through each of the calls that have to be canceled, let's use that same loop to mark as canceled the call as soon as we notify its callback.

The rationale is to prevent an issue happening on callback number 12 from avoiding canceled call number 2 from being marked as canceled in the DB.